### DISCUSSION TITLE
classifier-free guidance

### DISCUSSION THREAD

**Discussion by kjhenner** [archinetai/audio-diffusion-pytorch#25] [General]
I notice that there's a UNetConditional1d class that seems to support classifier-free guidance. I'm curious if you've run any experiments or had any success with this.

Specifically, I've been thinking about using a large dataset I'm collecting (~100k two minute samples) with a handful of genre labels to train a guided model. It looks like the current module is intended to be used with token embeddings, but perhaps it could be adapted to use a one-hot vector of the genre label?

**Comment by flavioschneider** [comment]
It is working now after fixing the bug in [v0.0.66](https://github.com/archinetai/audio-diffusion-pytorch/releases/tag/v0.0.66)! I did a few successful experiments with text conditioning very recently. I've also heard of others doing experiments with one-hot data to embeddings (without the pretrained T5 text model) which have also been successful. For that, it seems that a cfg `embedding_scale` of 10-20 works better to align the sound to the correct labels.

**Reply by jbmaxwell** [reply]
Is there a guide or some sample code to get started on training for text-conditioning?
